pkg/token/manualtoken_test.go

```diff
@@ -0,0 +1,103 @@
+package token
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+func TestNewManualToken(t *testing.T) {
+	testCases := []struct {
+		name        string
+		oAuthConfig adal.OAuthConfig
+		clientID    string
+		resourceID  string
+		tenantID    string
+		token       *adal.Token
+		wantErr     bool
+	}{
+		{
+			name:        "valid inputs",
+			oAuthConfig: adal.OAuthConfig{},
+			clientID:    "test-client-id",
+			resourceID:  "test-resource-id",
+			tenantID:    "test-tenant-id",
+			token:       &adal.Token{AccessToken: "test-access-token"},
+			wantErr:     false,
+		},
+		{
+			name:        "nil token",
+			oAuthConfig: adal.OAuthConfig{},
+			clientID:    "test-client-id",
+			resourceID:  "test-resource-id",
+			tenantID:    "test-tenant-id",
+			token:       nil,
+			wantErr:     true,
+		},
+		{
+			name:        "empty client ID",
+			oAuthConfig: adal.OAuthConfig{},
+			clientID:    "",
+			resourceID:  "test-resource-id",
+			tenantID:    "test-tenant-id",
+			token:       &adal.Token{AccessToken: "test-access-token"},
+			wantErr:     true,
+		},
+		{
+			name:        "empty resource ID",
+			oAuthConfig: adal.OAuthConfig{},
+			clientID:    "test-client-id",
+			resourceID:  "",
+			tenantID:    "test-tenant-id",
+			token:       &adal.Token{AccessToken: "test-access-token"},
+			wantErr:     true,
+		},
+		{
+			name:        "empty tenant ID",
+			oAuthConfig: adal.OAuthConfig{},
+			clientID:    "test-client-id",
+			resourceID:  "test-resource-id",
+			tenantID:    "",
+			token:       &adal.Token{AccessToken: "test-access-token"},
+			wantErr:     true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			provider, err := newManualToken(tc.oAuthConfig, tc.clientID, tc.resourceID, tc.tenantID, tc.token)
+
+			if tc.wantErr {
+				if err == nil {
+					t.Errorf("Expected error, but got nil")
```

Comment:
like i said earlier, it's better to verify the error. in that case, it's better to log the expected error and the returned error. for instance: `t.Errorf("expected error %v, but got %v", tc.wantErr, err)`